lib/Tumblr/StreamBuilder/Streams/ProportionalRoundRobinStream.php

```diff
@@ -159,7 +160,9 @@ protected function _enumerate(
                 $minor_stream_result =
                     $stream->enumerate($ct, $cursor->get_minor_stream_cursor($stream_index), $tracer, $option);
             } catch (\Exception $e) {
-                // swallow minor stream's exception and keep enumerate main stream
+                $log = StreamBuilder::getDependencyBag()->getLog();
+                $log->exception($e, $this->get_identity());
+                // keep enumerate main stream
```

Comment:
```suggestion
                // keep enumerating the stream despite the exception
```

Reply:
makes me wonder whether we should have some kind of configuration option to swallow or throw exceptions in cases like this... doesn't have to be for this PR, that's a bigger architecture question